### ISSUE TITLE
"Upgrade connections to HTTPS" (aka HTTPS Everywhere) causes significant page load delay

### ISSUE THREAD

**Issue by sybercorp** [brave/brave-browser#6196]
## Description 
Since the last few public full releases (not nightly or dev builds), there has been an issue with page loading being stuck in the "processing request" phase for extremely long periods of time (roughly 30 seconds) before finally displaying a page.  After a bit of troubleshooting, I have found the cause is the "Upgrade connections to HTTPS" option (aka. HTTPS Everywhere).  It happens if you use either the built-in option or the official HTTPS Everywhere plugin, so I'm not sure which is really to blame (the browser for its implementation or the rulebase EFF).  It seems to be limited to sites that automatically redirect from HTTP to HTTPS on their own anyway (without the need for a conversion in the browser or via plugin.  If you go directly to the HTTPS version of these sites, there is no delay at all.  If you disable the plugin or the feature to redirect from HTTP to HTTPS (leaving it to the site itself to redirect users) there is no delay at all.

## Steps to Reproduce

   1. Enable "Upgrade connections to HTTPS" or have the HTTPS Everywhere plugin installed/enabled
   2. Browse to a site that redirects from HTTP to HTTPS anyway (i.e., http://spotify.com)
   3. Notice the delay in page loading

## Actual result:
Roughly 30 seconds delay in page loading while the site switches from HTTP to HTTPS.

## Expected result:
No delay in redirecting users from HTTP to HTTPS

## Reproduces how often: 
100% reproducible

## Brave version (brave://version info)
Version 0.68.141 Chromium: 77.0.3865.90 (Official Build) (64-bit)

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? Haven't tested
- Can you reproduce this issue with the dev channel? Haven't tested
- Can you reproduce this issue with the nightly channel? Haven't tested

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? Yes
- Does the issue resolve itself when disabling Brave Rewards? No
- Is the issue reproducible on the latest version of Chrome? No

**Comment by sybercorp** [comment]
This appears to be fixed in the 0.68.142 release (it was present in the 0.68.141 release), though the release notes mention nothing about this issue or anything related to it being addressed.

**Comment by sybercorp** [comment]
This is still present in 0.69.132.

**Comment by bsclifton** [comment]
@diracdeltas any tips for debugging?

**Comment by sybercorp** [comment]
@bsclifton @diracdeltas The strangest part about this issue is that it doesn't happen on ALL sites that redirect from HTTP to HTTPS - only some.  For example, going to sites like wholefoods.com redirects immediately from HTTP to HTTPS, but other sites like nytimes.com has a significant delay before redirecting.  The delay goes away as soon as you disable the HTTPS Upgrade feature (and clear cache/history), then try again.  Also, I have not been able to recreate the issues in the macOS or Android builds, so it seems isolated to the Windows build.  The macOS and Windows setups are set identically, with the same extensions and such in them, for the record.

**Comment by rebron** [comment]
cc: @AndriusA @too4words

**Comment by sybercorp** [comment]
This still happens in 0.70.121.  I just tested yahoo.com, nytimes.com, and bing.com.  All of them hung for over a minute before loading, with HTTPS Everywhere enabled.  As soon as I disabled it and tried again, they load almost instantly.

**Comment by AndriusA** [comment]
@sybercorp that does indeed sound very strange, I've checked with both macOS and Windows 10 (beta and current stable 0.70.121), explicitly specifying `http://` and `https://`  for the three sites and still failing to reproduce...

Is there anything on your network that might be intercepting HTTPS traffic?

**Comment by sybercorp** [comment]
@AndriusA It's network agnostic (I can recreate it no matter where I am).  And, no, if that were the case, it wouldn't work at all when I went to HTTPS sites.  The problem is that the HTTPS Everywhere (Upgrade Connections to HTTPS) feature is interfering with the redirection/URL-rewrite from HTTP to HTTPS for certain sites (which makes some sense since HTTPS Everywhere works on a rulebase for specific sites, not just globally for all sites).  I'll record my scree then upload the video to YouTube, to show you all what it looks like with HTTPS Everywhere enabled versus disabled.

Follow-up comment with the links coming soon.

**Comment by sybercorp** [comment]
Here are the links to the videos I made.

Broken (HTTPS Everywhere enabled): https://youtu.be/-xsgNWqu55o
Working (HTTPS Everywhere disabled): https://youtu.be/Ko_ooDZPWmg

**Comment by sybercorp** [comment]
Is anyone going to offer any help with this?  Most sites redirect from HTTP to HTTPS on their own, so leaving HTTPS Everywhere disabled isn't really too big of a deal these days, but there's still a bug and I provided videos as proof to help you guys.  To date, I see there have been 0 views of those videos I provided, which is disheartening since you have only said you were unable to recreate the issue and see it in action, so I made the videos to show you the issue in action.

**Comment by sybercorp** [comment]
@rebron Does you removing this from Untriaged Backlog mean that you aren't going to fix this at all?

**Comment by jonathansampson** [comment]
@sybercorp Sorry for the radio-silence; would you be able to run a few more tests for me?

1. Open a Guest Window and see how long the navigation takes
2. If possible, connect to a mobile hotspot and see if this has any effect


Please do let us know what results you find.

**Comment by sybercorp** [comment]
@jonathansampson 
1. I don't have Windows installed any longer, so I can't test that.
2. I tried that  as part of what I mentioned before (in https://github.com/brave/brave-browser/issues/6196#issuecomment-546508853), that it doesn't matter which network I'm connected to.

**Comment by jonathansampson** [comment]
@sybercorp Was this issue only happening on Windows? If not, we can test on Linux/macOS as well. Do you run any type of antivirus software in parallel and/or VPN software on the device?

**Comment by sybercorp** [comment]
> @sybercorp Was this issue only happening on Windows? If not, we can test on Linux/macOS as well. Do you run any type of antivirus software in parallel and/or VPN software on the device?

Yes it was isolated to Windows (see my previous comment https://github.com/brave/brave-browser/issues/6196#issuecomment-540020246).

There is VPN software installed but whether it was in use or not had no bearing on the issue.

AV was installed but had none of its browser or network detection modules enabled or installed.

**Comment by jonathansampson** [comment]
@sybercorp Thank you for the prompt response! Unfortunately, I don't believe we have a clear test path now that you are no longer on Windows. I will keep my eyes and ears open for similar reports from other users, and merge with this if/when they arise. Thank you again for the feedback, and please do let us know if there is anything further we can do for you.

**Comment by jonathansampson** [comment]
Closing as stale.
